backend/project/endpoints/projects/groups/group_student.py

```diff
@@ -0,0 +1,103 @@
+"""Endpoint for joining and leaving groups in a project"""
+
+
+from os import getenv
+from urllib.parse import urljoin
+from dotenv import load_dotenv
+from flask import request
+from flask_restful import Resource
+from sqlalchemy.exc import SQLAlchemyError
+
+from project.utils.query_agent import insert_into_model
+from project.models.group import Group
+from project.utils.authentication import login_required_return_uid
+
+from project import db
+
+load_dotenv()
+API_URL = getenv("API_HOST")
+RESPONSE_URL = urljoin(f"{API_URL}/", "groups")
+
+
+class GroupStudent(Resource):
+    """Api endpoint to allow students to join and leave project groups"""
+    @login_required_return_uid
+    def post(self, project_id, group_id, uid=None):
+        """
+        This function will allow students to join project groups if not full
+        """
+        group = db.session.query(Group).filter_by(
```

Comment:
Check somewhere whether student is a student of the course, could be an authentication function